magma/magma
# chore: magma deb build is bazelified

## Summary

This change introduces a first throw for a magma debian build with bazel.
* this approach tries to be as near as possible to the make build - but there are differences
  * especially, the services are not packaged hermetically, i.e., common dependencies are only packaged once
* far less dependencies are needed because bazel builds as hermetic as possible

### known issues
* the resulting package is very large `(~900mb)` and with `--config=production` even larger
* there is some debug logging (basically warnings) during a build because files are overridden during packaging - these should only be generated proto files that are used by different services

## Test Plan

* build magma debian package on VM [edit] ~~or bazelBase/devcontainer~~ (the bazelBase/devcontainer sessiond services is currently not compatible with magma_deb)
  * `bazel build //lte/gateway/release:magma_deb_pkg`
  * add `--config=production` to be closer to the make debian build
  * ~~note: in rebuild the vscode container setup because of recent needed changes (#13743)~~
* copy the artifact to an accessible place[edit]: `~/magma$ cp bazel-bin/lte/gateway/release/magma_1.8.0_amd64.deb .`
* in `lte/gateway/deploy/roles/magma_deploy/tasks/main.yml` modify "Installing magma."
  * `# name: "{{ packages }}"`
  * add `deb: "/home/vagrant/magma/magma_1.8.0_amd64.deb"`
* run `~/magma/lte/gateway$ vagrant up magma_deb`
* check `magma_deb` for issues, services running, logging, ...
* integration tests can be run vs `magma_deb`

## Additional Information
